numpy/core/src/private/lowlevel_strided_loops.h

```diff
@@ -701,9 +701,12 @@ npy_bswap8_unaligned(char * x)
                         )
 
 #define PyArray_TRIVIAL_PAIR_ITERATION_STRIDE(size, arr) ( \
-                        size == 1 ? 0 : ((PyArray_NDIM(arr) == 1) ? \
-                                          PyArray_STRIDE(arr, 0) : \
-                                          PyArray_ITEMSIZE(arr)))
+                size == 1 ? 0 : ((PyArray_NDIM(arr) == 1) ? \
+                                  PyArray_STRIDE(arr, 0) : \
+                                  PyArray_ITEMSIZE(arr))); \
+                assert((PyArray_NDIM(arr) <= 1) || \
+                       (PyArray_FLAGS(arr) & \
```

Comment:
Would be clearer to us `PyArray_IS_F_CONTIGUOUS` and `PyArray_IS_C_CONTIGUOUS`. Should be just as efficient after the compiler gets through with it.

Reply:
Or `PyArray_CHKFLAGS`.

Reply:
PyArray_CHECKFLAGS which the macro uses is a function, but I guess that gets inlined anyway, half was that the other half was the surroundings does it mostly manual. But can change.

Reply:
The functions are actually preferred, although the structure member hiding project has stalled.

Reply:
I don't think it is very important, but keeping down the levels of indentation helps in reading the code.

Reply:
Why not just spell it `PyArray_TRIVIALLY_ITERABLE(arr)`?